src/utils/hooks/useLoadApplication.tsx

```diff
@@ -68,6 +71,15 @@ const useLoadApplication = ({ serialNumber, networkFetch }: UseGetApplicationPro
 
       setApplication(applicationDetails)
 
+      const { id, code, name, startMessage } = application.template as Template
+
+      setTemplateType({
+        id,
+        code,
+        name: name as string,
+        startMessage: startMessage ? startMessage : undefined,
+      })
+
```

Comment:
Just wondering why you only want these four fields? Why not just return the whole `application.template` object, even if it's not all used currently?

Also, just a question: where was the PageWrapper getting this info from previously?

Reply:
This info was not required for PageWrapper, and it's only now to load the `ApplicationStart` component. Before the component was called by `ApplicationCreate` and this information was getting retrieved by `useLoadTemplate`

Reply:
I don't like so much the idea of returning the Template object because it has so many unusable fields in it, that's why I'm always converting into a local type. I think this can stay the same way